addons/binding/org.openhab.binding.zwave/src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveMultiAssociationCommandClass.java

```diff
@@ -282,22 +284,23 @@ public SerialMessage setAssociationMessage(int group, int node, int endpoint) {
      * @return the serial message
      */
     public SerialMessage removeAssociationMessage(int group, int node, int endpoint) {
-        logger.debug("NODE {}: Creating new message for command MULTI_ASSOCIATIONCMD_REMOVE", getNode().getNodeId());
+        logger.debug(
+                "NODE {}: Creating new message for command MULTI_ASSOCIATIONCMD_REMOVE node {}, endpoint {}, group {}",
+                getNode().getNodeId(), node, endpoint, group);
         SerialMessage result = new SerialMessage(getNode().getNodeId(), SerialMessageClass.SendData,
                 SerialMessageType.Request, SerialMessageClass.SendData, SerialMessagePriority.Set);
 
-        byte[] newPayload;
-        if (endpoint == 0) {
-            logger.trace("NODE {}: Endpoint is 0. Sending only node.", getNode().getNodeId());
-            newPayload = new byte[] { (byte) getNode().getNodeId(), 4, (byte) getCommandClass().getKey(),
-                    (byte) MULTI_ASSOCIATIONCMD_REMOVE, (byte) (group & 0xff), (byte) (node & 0xff) };
-        } else {
-            logger.trace("NODE {}: Endpoint not 0. Sending node and endpoint.", getNode().getNodeId());
```

Comment:
Endpoint 0 can be set in the arguments and is then handled directly in the command.